### ISSUE TITLE
Streams

### ISSUE THREAD

**Issue by SoniEx2** [TokTok/c-toxcore#1270]
When talking to an user, most chat apps expect you to maintain a single conversation with that user. However, this is often not the case, and it's possible to talk about multiple things with the same user.

This comes with the caveat that any logs become hard to follow. As such, it would maybe be a good idea to split the logs based on what I call "streams".

It's easier to follow different conversations with the same person on e.g. IRC if you use different channels for each of them. Streams are like IRC channels, but they go between the same two parties, always. Either party can open a stream, or close it, and it's limited to the 2 parties.

The main benefit is with logging, where each stream would have a separate log/buffer. If you were using tox from a pager, streams wouldn't really matter much.

I also believe streams should be named, like IRC channels. This means it's also possible to reopen a stream and append to its log.

**Comment by notramo** [comment]
https://zulipchat.com/ already does this. Maybe you can get some idea on how it should work from there.

**Comment by SoniEx2** [comment]
just attach a stream tag to messages and add some "join stream" message.

the other party should automatically join, and clients that don't support it will just put everything in the same chat.

note that this is specifically about direct messaging, NOT groups.

**Comment by notramo** [comment]
@SoniEx2 but the same thing could be applied ALSO to groups.

**Comment by SoniEx2** [comment]
in which case we don't want auto joining

or we want some way to restrict valid values

**Comment by zoff99** [comment]
we need more devs to implement it, what it will be called in not the most important issue.
if you can help us to implement msgV2 please contact us.

**Comment by iphydf** [comment]
This won't be done in toxcore, but in a higher level wrapper lib.
